sc3-sc3.R

```diff
@@ -0,0 +1,198 @@
+    c("-t", "--n-cores"),
+    action = "store",
+    default = NULL,
+    type = "integer",
+    help = "Number of threads/cores to be used in the user's machine."
+  ),
+  make_option(
+    c("-k", "--kmeans-nstart"),
+    action = "store",
+    default = NULL,
+    type = "integer",
+    help = "nstart parameter passed to kmeans function. Default is 1000 for up to 2000 cells and 50 for more than 2000 cells."
+  ),
+    make_option(
+    c("-a", "--kmeans-iter-max"),
+    action = "store",
+    default = 1e+09,
+    type = "double",
+    help = "iter.max parameter passed to kmeans function. Default is 1e+09."
+  ),
+    make_option(
+    c("-g", "--k_estimator"),
+    action = "store",
+    default = FALSE,
+    type = "logical",
+    help = "A boolean variable which defines whether to estimate an optimal number of clusters. If user has already defined the ks parameter the estimation does not affect the user's paramater."
+  ),
+    make_option(
+    c("-b", "--biology"),
+    action = "store",
+    default = FALSE,
+    type = "logical",
+    help = "A boolean variable which defines whether to compute differentially expressed genes, marker genes and cell outliers."
+  ),
+    make_option(
+    c("-s", "--rand-seed"),
+    action = "store",
+    default = 1,
+    type = "integer",
+    help = "sets the seed of the random number generator. SC3 is a stochastic method, so setting the rand_seed to a fixed values can be used for reproducibility purposes."
+  ),
+  make_option(
+    c("-o", "--output-object-file"),
+    action = "store",
+    default = NA,
+    type = "character",
+    help = "File name in which to store serialized R object of type 'SingleCellExperiment'."
+  )
+)
+
+opt <- wsc_parse_args(option_list, mandatory = c("input_object_file", "output_object_file"))
+
+# Check parameter values defined
+if ( ! file.exists(opt$input_object_file)){
+  stop(paste("File object or matrix", opt$input_object_file, "does not exist"))
+}
+
+# See if the user has specified a file for svm_train_inds, in which case we need to read it and create a vector
+
+if (! is.null(opt$svm_train_inds)){
+  if (! file.exists(opt$svm_train_inds)){
+    stop(paste("Supplied svm_train_inds file", opt$svm_train_inds, "does not exist"))
+  }else{
+    svm_train_inds <- readLines(opt$svm_train_inds)
+  }
+}else{
+  svm_train_inds <- NULL
+}
+
+
+# Once arguments are satisfcatory, load SC3 package
+suppressPackageStartupMessages(require(SC3))
+suppressPackageStartupMessages(require(SingleCellExperiment))
+
+# Input from serialized R object
+SingleCellExperiment <- readRDS(opt$input_object_file)
+
+# SC3 needs this
+rowData(SingleCellExperiment)$feature_symbol <- rownames(SingleCellExperiment)
+
+# Temporary fix to deal with https://github.com/hemberg-lab/SC3/issues/53
+assignInNamespace(x = "rowSums", value = Matrix::rowSums, ns = "base")
+
+# Create SCE object from data
+## If the user has specified ks, parse the ks to a vector and use as ks value
+if (! is.null(opt$ks)){
```

Comment:
As far as I can see the only difference between these duplicated sc3() calls is the value of ks. So could you move that call to a single one after the conditional please? Only the ks setting needs to be conditional.